### PR TITLE
Sidm 5013 step up authentication

### PR DESCRIPTION
**Pull request by radoslaw-orlowski-amido**
### Change description ###

StepUp authn.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    protected String getSessionToken(@Nonnull final HttpServletRequest request) {
+        return Arrays.stream(getCookiesFromRequest(request))
+            .filter(cookie -> idamSessionCookieName.equals(cookie.getName()))
+            .map(Cookie::getValue)
+            .findAny()
+            .orElseThrow();
+    }
+
+    protected boolean isAuthorizeRequest(@Nonnull final HttpServletRequest request) {
+        return request.getRequestURI().contains("/o/authorize") && "post".equalsIgnoreCase(request.getMethod());
```
Comment (kremi) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java:110]: Normally, the Authorization request is `GET /o/authorize`. Are we sure we need a `post` here?

Comment (radoslaw-orlowski-amido) [reply src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java]: Fixed

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    protected boolean isAuthorizeRequest(@Nonnull final HttpServletRequest request) {
+        return request.getRequestURI().contains(OIDC_AUTHORIZE_ENDPOINT) &&
+            ("post".equalsIgnoreCase(request.getMethod()) || "get".equalsIgnoreCase(request.getMethod()));
+    }
+
+    protected boolean hasSessionCookie(@Nonnull final HttpServletRequest request) {
+        return Arrays.stream(getCookiesFromRequest(request)).anyMatch(cookie -> idamSessionCookieName.equals(cookie.getName()));
+    }
+
+    protected void copyRequestParameters(@Nonnull final URIBuilder uriBuilder, @Nonnull final HttpServletRequest originalRequest) {
```
Comment (kremi) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java:116]: This method is no longer needed.

**Review by kremi — APPROVED** [review]
Great work 👍

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java (hunk 2 above)
Comment (radoslaw-orlowski-amido) [reply src/main/java/uk/gov/hmcts/reform/idam/web/strategic/StepUpAuthenticationZuulFilter.java]: Deleted

**Pull request merged**